andela/Ah-backend-aquaman
# #164069223 Create articles

#### What does this PR do?
This PR enables users to create, read, update and delete articles
#### Description of Task to be completed?
- Create an article
- View all articles
- View a single article
- Edit an article
- Delete an article

#### How should this be manually tested?

* Clone the repo, cd into Ah-backend-aquaman and checkout the branch ft-create-articles-64069223
* make a virtual environment and activate it
* pip install -r requirements.txt
* python manage.py makemigrations
* python manage.py migrate
* python manage.py runserver
* create an account,verify email and login


__Endpoints:__
-  post article `POST /api/articles/`
- get all articles `GET /api/articles/`
- get a single article `GET /api/articles/<slug>/`
- edit an article  `PATCH /api/articles/<slug>/`
- delete an article `DELETE /api/articles/<slug>/`


#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?

[#164069223](https://www.pivotaltracker.com/story/show/164069223)

#### Screenshots (if appropriate)
``` - Create article ```
<img width="1126" alt="Screenshot 2019-03-13 at 14 40 18" src="https://user-images.githubusercontent.com/20795487/54276625-d445bc00-459e-11e9-9ad1-ecd2c1bac664.png">
``` - get all articles```
<img width="1125" alt="Screenshot 2019-03-13 at 14 41 16" src="https://user-images.githubusercontent.com/20795487/54276706-08b97800-459f-11e9-8c25-207a84defd5f.png">

``` - get single article```
<img width="1133" alt="Screenshot 2019-03-13 at 14 42 01" src="https://user-images.githubusercontent.com/20795487/54276721-1838c100-459f-11e9-87e1-807dd48e1770.png">

``` - update an article```
<img width="1126" alt="Screenshot 2019-03-13 at 14 43 21" src="https://user-images.githubusercontent.com/20795487/54276746-2981cd80-459f-11e9-88e4-96f5e4dec85f.png">
``` - delete an article```
<img width="1122" alt="Screenshot 2019-03-13 at 14 44 22" src="https://user-images.githubusercontent.com/20795487/54276764-34d4f900-459f-11e9-8870-7cb02bafbe07.png">
